yapapi/executor/__init__.py

```diff
@@ -528,14 +529,14 @@ async def get_batch_results() -> List[events.CommandEvent]:
                     future_results = loop.create_future()
                     results = await get_batch_results()
                     future_results.set_result(results)
-                    item = await command_generator.asend(future_results)
-                except StopAsyncIteration:
-                    raise
-                except Exception:
+                except Exception as e:
                     # Raise the exception in `command_generator` (the `worker` coroutine).
                     # If the client code is able to handle it then we'll proceed with
                     # subsequent batches. Otherwise the worker finishes with error.
                     item = await command_generator.athrow(*sys.exc_info())
+                else:
+                    item = await command_generator.asend(future_results)
+
```

Comment:
We shouldn't handle errors raised in `command_generator.asend()` as they include exceptions thrown in the worker function.